src/pointer.rs

```diff
@@ -845,6 +981,153 @@ const fn validate(value: &str) -> Result<&str, ParseError> {
     Ok(value)
 }
 
+/*
+░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+╔══════════════════════════════════════════════════════════════════════════════╗
+║                                                                              ║
+║                                  ParseError                                  ║
+║                                 ¯¯¯¯¯¯¯¯¯¯¯¯                                 ║
+╚══════════════════════════════════════════════════════════════════════════════╝
+░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+*/
+
+/// Indicates that a `Pointer` was malformed and unable to be parsed.
+#[derive(Debug, PartialEq)]
+pub enum ParseError {
+    /// `Pointer` did not start with a backslash (`'/'`).
+    NoLeadingBackslash,
+
+    /// `Pointer` contained invalid encoding (e.g. `~` not followed by `0` or
+    /// `1`).
+    InvalidEncoding {
+        /// Offset of the partial pointer starting with the token that contained
+        /// the invalid encoding
+        offset: usize,
+        /// The source `InvalidEncodingError`
+        source: InvalidEncodingError,
+    },
+}
+
+impl fmt::Display for ParseError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::NoLeadingBackslash { .. } => {
+                write!(
+                    f,
+                    "json pointer is malformed as it does not start with a backslash ('/')"
+                )
+            }
+            Self::InvalidEncoding { source, .. } => write!(f, "{source}"),
+        }
+    }
+}
+
+impl ParseError {
+    /// Returns `true` if this error is `NoLeadingBackslash`; otherwise returns
+    /// `false`.
```

Comment:
You need this to be a single line for rustdoc to parse it properly. Otherwise it'll render weirdly. Ditto elsewhere (picked this location semi-randomly).

Reply:
Really? Is there a flag or something that I need to enable for it to respect markdown newlines on docs.rs? I thought I had them all enabled locally. If not, I think it respects it:
<img width="1002" alt="Screenshot 2024-07-08 at 4 07 09 PM" src="https://github.com/chanced/jsonptr/assets/1073892/25fa748a-3cf1-4259-93d5-c25e0f6cde23">

Reply:
Hmmm so when I ask `rust-analyzer` for inlay docs it shows me a pop-up with a hard break at the newline. Doesn't seem to render the second line differently, so not sure how much of a concern this is really, but Jon once said this could cause issues in places as the first line is indexed specially.

Reply:
Oh that's unfortunate. I use a plugin to linebreak my comments and I guess I shouldn't do that anymore. I had no idea that any tooling in the ecosystem treated newlines as hard breaks. I've always assumed it respected markdown rules.

btw, I'm reluctant to respond knowing you're tired. Go get some rest. This can wait.

Reply:
Aw man, I have so many comments to fix. I'm more bummed that I can't auto-format my doc comments anymore. 

Strangely enough, vscode gets it right. 
<img width="642" alt="Screenshot 2024-07-08 at 4 35 24 PM" src="https://github.com/chanced/jsonptr/assets/1073892/005774ec-a531-4d29-ad94-5c08f686e32b">

Btw, is that neovim? 

> Jon also said is that the first line "shows up in more places", though I never figured out which places those are, exactly. 😅

lol!

Reply:
I think this one's okay. It seems to be a bug in RA not properly rendering newlines. Which is weird, you'd think that'd have been caught by now.

I wonder what else could possibly be incorrectly parsing markdown and only respecting the first actual line rather than the first paragraph.

Reply:
I wish there was a way to turn an entire discussion into an issue. I've update some of them, but not all.

Reply:
https://github.com/chanced/jsonptr/issues/58